Mathlib/RingTheory/Smooth/StandardSmooth.lean

```diff
@@ -0,0 +1,229 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jung Tao Cheng, Christian Merten, Andrew Yang
+-/
+import Mathlib.Algebra.MvPolynomial.PDeriv
+import Mathlib.LinearAlgebra.Determinant
+import Mathlib.RingTheory.Presentation
+
+/-!
+# Standard smooth algebras
+
+In this file we define standard smooth algebras. For this we introduce
+the notion of a `PreSubmersivePresentation`. This is a presentation `P` that has
+fewer relations than generators. More precisely there exists an injective map from `P.rels`
+to `P.vars`. To such a presentation we may associate a jacobian. `P` is then a submersive
+presentation, if its jacobian is invertible.
+
+Finally, a standard smooth algebra is an algebra that admits a submersive presentation.
```

Comment:
Please include (here or in the TODO, or both) a discussion of how this notion relates to the "other" definition of smooth.